Dockerfile

```diff
@@ -14,30 +14,33 @@
 
 ##
 # Build stage
+# Use github action runner cached images to avoid being rate limited
+# https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md#cached-docker-images
 ## 
-FROM rust:1-buster AS build-env
+FROM buildpack-deps:buster AS build-env
 
 # Install dependencies
-RUN apt-get update && apt-get install --no-install-recommends -y build-essential clang ocl-icd-opencl-dev cmake
+RUN apt-get update && apt-get install --no-install-recommends -y build-essential clang ocl-icd-opencl-dev cmake ca-certificates curl
+RUN update-ca-certificates
+
+# Install rustup
+# https://rustup.rs/
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+ENV PATH="/root/.cargo/bin:${PATH}"
+ENV RUSTFLAGS="-Ctarget-feature=+avx2,+fma"
 
 WORKDIR /usr/src/forest
 COPY . .
 
-# Grab the correct toolchain
-RUN rustup toolchain install nightly
-
-ENV RUSTFLAGS="-Ctarget-feature=+avx2,+fma"
-
 # Install Forest
 RUN make install
 
-# strip symbols to make executable smaller
-RUN strip /usr/local/cargo/bin/forest /usr/local/cargo/bin/forest-cli
-
 ##
 # Prod image for forest binary
+# Use github action runner cached images to avoid being rate limited
+# https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2004-Readme.md#cached-docker-images
 ##
-FROM debian:10-slim
+FROM debian:10
```

Comment:
I'm not sure it's a proper way out of this. Prod image should be as small as possible, and Debian is relatively large. Should we perhaps try `alpine`? We can od it in another task though.

Reply:
That's a good idea, but we don't have cached buildpack for alpine, better to investigate in another task

Reply:
For now let me switch to `ubuntu:20.04` whose size is similar to `debian:buster-slim`

Reply:
Great, thanks!